gen.go

```diff
@@ -118,6 +122,18 @@ func (g *Generator) With(parts ...string) *Generator {
 	}
 }
 
+func (g *Generator) WithGeneratedHeader() *Generator {
```

Comment:
Maybe we can change API a bit, and create a `WithOpts` method for applying different options to `Generator` struct?